### PR TITLE
[Documentation] 2.2-Install Configuration & 4.0-scripts

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
based off of #219. I shuffled a bit the information from the Appendix section of #219 to move to their more logical location.
~~Pull request for now only has placeholders for the section I am editing.~~

### CODE REVIEW

File: docs/02-Install.md [hunk 1]
```diff
@@ -1,13 +1,140 @@
...
+Under the section `Paths`
+ * `imagePath`
+
+**NEED TO TALK ABOUT THE CREATION OF NIFTI OPTIONS AS IT IS REFERENCED IN THE
+  PIPELINE FLOW DOCUMENTATION.**
+
+### 2.2.2 LORIS
+
+
+### 2.2.3 LORIS-MRI
```
Comment (MounaSafiHarab) [inline docs/02-Install.md]: add a note here about the environment file and specifically, the tmp_dir where all temporary processing happens in case projects want to customize this (default is /tmp)

**Comment by MounaSafiHarab** [comment]
recycled as much as I can from the wiki.

@cmadjar 
You can start giving feedback if you wish...

File: docs/02-Install.md [hunk 2]
```diff
@@ -1,13 +1,312 @@
...
+
+
+Projects can upload scans and launch the pipeline in a variety of options 
+detailed in [PipelineOptions](PipelineOptions.md). 
+Irrespective of the project's choice as to whether the imaging scan is to be 
+uploaded through the Imaging Uploader GUI or not, pipeline insertion progress 
+can be consulted through a live 'Log Viewer' panel.
+Some settings need to be configured properly (`php.ini` variables, 
+`MRI-Upload Directory` and `ImagingUploader Auto Launch`), and are documented in 
+the [LORIS repository: Imaging Uploader Specification](https://github.com/aces/Loris/blob/master/modules/imaging_uploader/README.md).
```
Comment (MounaSafiHarab) [inline docs/02-Install.md]: these links point now to master and they are only on minor branch, but eventually when minor is released it should be fine.

File: docs/04-Scripts.md [hunk 3]
```diff
@@ -1 +1,145 @@
...
+=head1 AUTHORS
+
+LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative 
+Neuroscience 
+
+=cut
+
+</pre>
+
+## 4.2 - Common insertion scripts re-run
```
Comment (MounaSafiHarab) [inline docs/04-Scripts.md]: Here I migrate from the wiki sections that are scripts related when a user wants to re-run partially some scripts...

File: docs/AppendixA-FAQ.md [hunk 4]
```diff
@@ -1 +1,69 @@
 # Appendix A - FAQ
+
```
Comment (MounaSafiHarab) [inline docs/AppendixA-FAQ.md]: these remaining sections; not sure where it makes most sense to move them to...

File: docs/02-Install.md [hunk 5]
```diff
@@ -1,13 +1,312 @@
...
+the columns defining expected scan parameters (e.g. `TE_Range`) are defined very 
+broadly.  
+The `Scan_type` column values are defined in the `mri_scan_type` table 
+(e.g. 44=t1); do not include hyphens, spaces or periods in your 
+`mri_scan_type.Scan_type` column values.
+
+5. **`Config`** table
+
+The `Config` table can also be accessed and customized from the LORIS front-end, 
+under the Admin menu, the `Configuration` module. Here are the configuration
```
Comment (cmadjar) [inline docs/02-Install.md]: Rephrase 
"The Config table can also be accessed and customized from the LORIS front-end, under the Admin menu, the Configuration module. "
for 
"The Config table can also be accessed and customized from the LORIS front-end via the Configuration module accessible under the Admin menu. "

File: docs/02-Install.md [hunk 6]
```diff
@@ -1,13 +1,312 @@
...
+settings that impact directly or indirectly the pipeline:
+
+Under the section `Study`:
+ * `ImagingUploader Auto Launch`
```
Comment (cmadjar) [inline docs/02-Install.md]: would add a description of what populating this field do

File: docs/02-Install.md [hunk 7]
```diff
@@ -1,13 +1,312 @@
...
+ 
+Under the section `Paths`
+ * `Imaging Data`
```
Comment (cmadjar) [inline docs/02-Install.md]: Same here, add a short description of what those are maybe

File: docs/02-Install.md [hunk 8]
```diff
@@ -1,13 +1,312 @@
...
+ * `MRI-Upload Directory`
+ * `MINC Files`
+ * `Images`
+ 
+ Under the section `Imaging Modules`
+  * `Patient ID regex`: Used by the DICOM Archive module to show/hide the PatientID info
+  * `Patient name regex`: Used by the DICOM Archive module to show/hide the Patient Name info
+  * `Lego phantom regex`: Used by the DICOM Archive module to show/hide the Patient Name info for phantoms
+  * `Living phantom regex`: Used by the DICOM Archive module to show/hide the Patient Name info for phantoms
+  * `Imaging Browser Tabulated Scan Types`: Used by Imaging Browser to decide what modalities to display the QC status on
```
Comment (cmadjar) [inline docs/02-Install.md]: maybe rephrase the description to Imaging Browser Tabulated Scan Types to:
"Used by the the Imaging Browser main page listing the different imaging sessions. This determines which modalities should have their QC status displayed in that listing."

File: docs/02-Install.md [hunk 9]
```diff
@@ -1,13 +1,312 @@
...
+     
+ Under the section `Imaging Pipeline`:
+ * `Loris-MRI Data Directory`: Typically `/data/$PROJECT/bin/mri/`
```
Comment (cmadjar) [inline docs/02-Install.md]: Isn't it supposed to be /data/$PROJECT/data?

File: docs/02-Install.md [hunk 10]
```diff
@@ -1,13 +1,312 @@
...
+ * `Study Name`: Prefix used for all filenames inserted into Imaging Browser and the `files` table
+ * `User to notify when executing the pipeline`: User email address used when notification is to be sent by the pipeline
```
Comment (cmadjar) [inline docs/02-Install.md]: Would say: "User email address **to be** used when"

File: docs/02-Install.md [hunk 11]
```diff
@@ -1,13 +1,312 @@
...
+ Under the section `Imaging Modules`
+  * `Patient ID regex`: Used by the DICOM Archive module to show/hide the PatientID info
+  * `Patient name regex`: Used by the DICOM Archive module to show/hide the Patient Name info
+  * `Lego phantom regex`: Used by the DICOM Archive module to show/hide the Patient Name info for phantoms
+  * `Living phantom regex`: Used by the DICOM Archive module to show/hide the Patient Name info for phantoms
+  * `Imaging Browser Tabulated Scan Types`: Used by Imaging Browser to decide what modalities to display the QC status on
+     
+ Under the section `Imaging Pipeline`:
+ * `Loris-MRI Data Directory`: Typically `/data/$PROJECT/bin/mri/`
+ * `Study Name`: Prefix used for all filenames inserted into Imaging Browser and the `files` table
```
Comment (cmadjar) [inline docs/02-Install.md]: Would say: Prefix **to be** used **in** all filenames inserted into **the `files` table and visible in the frontend via the Imaging Browser module**

File: docs/02-Install.md [hunk 12]
```diff
@@ -1,13 +1,312 @@
...
+ * `User to notify when executing the pipeline`: User email address used when notification is to be sent by the pipeline
+ * `Full path to get_dicom_info.pl script`: Typically `/data/$PROJECT/bin/mri/dicom-arhive/get_dicom_info.pl`
+ * `Horizontal pictures creation`
```
Comment (cmadjar) [inline docs/02-Install.md]: missing description for Horizontal pictures creation

File: docs/02-Install.md [hunk 13]
```diff
@@ -1,13 +1,312 @@
...
+ * `NIFTII file creation`: Project choice to enable for automated NIFTII file creation
```
Comment (cmadjar) [inline docs/02-Install.md]: Would rephrase the description to:
Use to enable or disable automated NIfTI file creation
(PS: typo in NIfTI)

File: docs/02-Install.md [hunk 14]
```diff
@@ -1,13 +1,312 @@
...
+ * `dcm2mnc binary to use when converting`
```
Comment (cmadjar) [inline docs/02-Install.md]: missing description here

File: docs/02-Install.md [hunk 15]
```diff
@@ -1,13 +1,312 @@
...
+ * `Path to Tarchives`
```
Comment (cmadjar) [inline docs/02-Install.md]: missing description

File: docs/02-Install.md [hunk 16]
```diff
@@ -1,13 +1,312 @@
...
+ * `Upload creation of candidates`: Creates candidates into LORIS if enabled
```
Comment (cmadjar) [inline docs/02-Install.md]: To remain consistent with my rephrasing above if you accept it:
Enable or disable candidate creation into LORIS when running the insertion pipeline

File: docs/02-Install.md [hunk 17]
```diff
@@ -1,13 +1,312 @@
...
+ * `Project batch management used`
```
Comment (cmadjar) [inline docs/02-Install.md]: missing description

File: docs/02-Install.md [hunk 18]
```diff
@@ -1,13 +1,312 @@
...
+ * `If site is used`
```
Comment (cmadjar) [inline docs/02-Install.md]: missing description

File: docs/02-Install.md [hunk 19]
```diff
@@ -1,13 +1,312 @@
...
+ * `Number of volumes in native DTI acquisitions`: Used by the DTIPrep pipeline
+ * `Scan type of native T1 acquisition`: Used by the DTIPrep pipeline
+ * `Max number of DTI rejected directions for passing QC`: Used by the DTIPrep pipeline
+ * `NIAK Path`: Used by the DTIPrep pipeline
+ * `Secondary QCed dataset`: Used by the DTIPrep pipeline
+
+
+### 2.2.2 LORIS
```
Comment (cmadjar) [inline docs/02-Install.md]: Stopped here. Will continue this afternoon.

Comment (MounaSafiHarab) [reply docs/02-Install.md]: thanks! will incorporate all your comments and let you know when I do, but probably not today or tomorrow (I had the luxury to dedicate a big part of yesterday to documentation, but now things piled up on the project and I need to catch up there!)

File: docs/02-Install.md [hunk 20]
```diff
@@ -1,13 +1,312 @@
...
+  [README.md](../README.md) file.
 
-## 2.3 Configuration
 
-## 2.4 Pipeline flow
+## 2.2 Configuration
 
-## 2.5 Common errors
+Following a successful install, some configurations and customizations are 
+needed on the database, LORIS and LORIS-MRI sides as outlined in the following
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md [hunk 21]
```diff
@@ -1,13 +1,312 @@
...
+
+2. **`Visit_Windows`** table
+
+To populate with visit labels, you can manually insert study-specific 
+information:
+
+
+      INSERT INTO Visit_Windows (Visit_label,  WindowMinDays, WindowMaxDays, OptimumMinDays, OptimumMaxDays, WindowMidpointDays) VALUES ('V1', '0', '100', '40', '60', '50');
+
+If age is a not critical factor in study visit scheduling, define Min value as
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md [hunk 22]
```diff
@@ -1,13 +1,312 @@
...
+2. **`Visit_Windows`** table
+
+To populate with visit labels, you can manually insert study-specific 
+information:
+
+
+      INSERT INTO Visit_Windows (Visit_label,  WindowMinDays, WindowMaxDays, OptimumMinDays, OptimumMaxDays, WindowMidpointDays) VALUES ('V1', '0', '100', '40', '60', '50');
+
+If age is a not critical factor in study visit scheduling, define Min value as 
+0, and Max value as 39999.
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: good question. I will being in @christinerogers on this. I got this from the wiki.
We actually in CCNA put this value: `2147483647`!!!

@ridz1208 do you know what is the correct number to put for Max Value in the case where Age is not critical?

File: docs/02-Install.md (hunk 5 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 6 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 7 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done; BUT I find many fields are super redundant; added an item to the imaging meeting agenda as I am sure we can group some

File: docs/02-Install.md (hunk 8 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done with a couple of edits

File: docs/02-Install.md (hunk 9 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: it sure should be :)

File: docs/02-Install.md (hunk 11 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok

File: docs/02-Install.md (hunk 13 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 16 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok

**Comment by MounaSafiHarab** [comment]
@cmadjar 
I did a first round of editing; I incorporated all your comments; except for a couple of missing descriptions, which I will incorporate as you go through more reading.

Merci!

File: docs/02-Install.md (hunk 17 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 18 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 15 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 14 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 12 above)
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: I just confirmed with @ridz1208 
he said this value should be set to Max int which is the value I show above.
@christinerogers maybe it is worth it to update the wiki accordingly?

File: docs/04-Scripts.md [hunk 23]
```diff
@@ -1 +1,145 @@
...
+
+##### Bypassing protocol violation checks
+
+For cases when a scan has triggered a protocol violation, the minc volume can be
+  **force-loaded** into LORIS by running:
+```
+uploadNeuroDB/minc_insertion.pl -acquisition_protocol t2w -bypass_extra_file_checks -create_minc_pics -profile prod -globLocation -force  -tarchivePath _/data/project/dataTransfer/library/2009/DCM_2009-09-25_project_20110214_185904581.tar_ -mincPath _/data/project/data/trashbin/TarLoad-3-34-pVzGC5/xxx0067_703739_v12_20090925_222403_18e1_mri.mnc_
```
Comment (cmadjar) [inline docs/04-Scripts.md]: I don't understand why there are _ around the file names in the command

File: docs/04-Scripts.md [hunk 24]
```diff
@@ -1 +1,145 @@
...
+See also: [MRI-PR#141](https://github.com/aces/Loris-MRI/pull/141) for more
+  examples.
+
+#### Rerunning the Imaging pipeline
+
+- If one of the final steps such as the MINC conversion is failing, you may
+    wish to just re-run the tarchiveLoader script.
+    
+> When the need arises to re-load imaging data in LORIS, it is generally not
+   sufficient to just re-run the MINC/NIFTII loading step (`tarchiveLoader` or
```
Comment (cmadjar) [inline docs/04-Scripts.md]: NIfTI instead of NIFTII

File: docs/04-Scripts.md [hunk 25]
```diff
@@ -1 +1,145 @@
...
+- `parameter_file`
+- `tarchive`
+- `mri_acquisition_dates`
+- `files` (best to delete from this table last)
+- `mri_upload`
+- `session` - not recommended - only if necessary, and only if no other data is
+    associated to this session e.g. on the Behavioural side of Loris.
+
+It is also recommended to remove from the tarchive directory the last generated
+  *.tar package for this dataset.
```
Comment (cmadjar) [inline docs/04-Scripts.md]: Should probably also remove the files in assembly, pic, jiv and trashbin directory?

File: docs/04-Scripts.md [hunk 26]
```diff
@@ -1 +1,145 @@
...
+
+For backing up, re-labelling and re-loading MRI datasets with QC information,
+  see [Beta Tutorial](https://github.com/aces/Loris/wiki/Reloading-MRI-data-for-mislabelled-session)
+
+#### Multiple scanner datasets per session
+
+In cases where a subject was scanned in two scanner sessions within a single
+  study Timepoint, both datasets should be loaded and associated to the same
+  visit label / session table record. Create separate tarchives for each
+  DICOM dataset upload each to the same visit.
```
Comment (cmadjar) [inline docs/04-Scripts.md]: The last sentence is confusing. Maybe rephrase?

File: docs/AppendixA-FAQ.md [hunk 27]
```diff
@@ -1 +1,69 @@
...
+- Depending on your operating system, some dependencies might be missing.
+    During initial troubleshooting of the imaging pipeline, note any related
+    error messages (e.g. `install_driver(mysql) failed: Can't locate DBD/mysql
+   .pm`) and install missing packages as needed (e.g.
+   `sudo apt-get install libdbi-perl`, `sudo apt-get install libdbd-mysql-perl`,
+    `sudo apt-get install libarchive-zip-perl`).
+
+- If your MINC toolkit is older than 1.9.14 and your scans have no Date of Birth
+   value, you may see an age unit error during DICOM to MINC conversion.
+   Instructions for compiling a more recent version available on
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: "...more recent version **of the MINC toolkit are** available on..."

File: docs/AppendixA-FAQ.md [hunk 28]
```diff
@@ -1 +1,69 @@
...
+   [MNI-BIC GitHub](https://github.com/BIC-MNI/minc-toolkit-v2).
+
+### A.2 Verify loaded images and Troubleshooting
+
+Once an MRI scan session has been successfully inserted, it will be listed in
+  the Imaging Browser main data table (also linked from the Dicom Archive
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: Dicom should be upper case

File: docs/AppendixA-FAQ.md [hunk 29]
```diff
@@ -1 +1,69 @@
...
+   .pm`) and install missing packages as needed (e.g.
+   `sudo apt-get install libdbi-perl`, `sudo apt-get install libdbd-mysql-perl`,
+    `sudo apt-get install libarchive-zip-perl`).
+
+- If your MINC toolkit is older than 1.9.14 and your scans have no Date of Birth
+   value, you may see an age unit error during DICOM to MINC conversion.
+   Instructions for compiling a more recent version available on
+   [MNI-BIC GitHub](https://github.com/BIC-MNI/minc-toolkit-v2).
+
+### A.2 Verify loaded images and Troubleshooting
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: Change title to something like: Images display trouble shooting notes

File: docs/AppendixA-FAQ.md [hunk 30]
```diff
@@ -1 +1,69 @@
...
+  imaging dataset -- consult the `mri_upload` database table to look up which
+  uploadID has been assigned to your scans.
+
+***Caveat:*** When the imaging insertion pipeline is auto-launched by the
+  Imaging Uploader module, the pipeline scripts' log files are output to
+  `/tmp/` and deleted. To avoid deletion, edit the Server Processes Manager
+  function [deleteProcessFiles()](https://github.com/aces/Loris/blob/master/modules/server_processes_manager/php/AbstractServerProcess.class.inc#L521)
+  to return false instead of true.
+
+#### Troubleshooting Insertion of uploaded datasets
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: I would tend to have that as ### A3. Insertion script troubleshooting notes

File: docs/AppendixA-FAQ.md [hunk 31]
```diff
@@ -1 +1,69 @@
...
+
+If upload was successful but issues were encountered with the imaging insertion
+  pipeline scripts:
+
+- CentOS: check for additional dependencies/configurations (e.g. Dicom
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: Dicom should be upper case

File: docs/AppendixA-FAQ.md [hunk 32]
```diff
@@ -1 +1,69 @@
...
+    Dictionary path) in the detailed
+    [CentOS Imaging Installation transcript](https://github.com/aces/Loris/wiki/CentOS-Imaging-installation-transcript)
+- Manually re-run the entire pipeline sequence:
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: Would probably rephrase  like this:

Manually re-run the entire pipeline sequence **using the** imaging_upload_file.pl **script**

File: docs/AppendixA-FAQ.md [hunk 33]
```diff
@@ -1 +1,69 @@
...
+    [imaging_upload_file.pl](#post-upload:-pre-processing-and-insertion-into-loris)
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: link to imaging_upload_file.pl does not show

File: docs/AppendixA-FAQ.md [hunk 34]
```diff
@@ -1 +1,69 @@
...
+- If one of the final steps such as the MINC conversion is failing, you may
+    wish to just re-run the tarchiveLoader script.
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: if there is a link to imaging_upload_file.pl, shouldn't there be one for tarchiveLoader?

File: docs/AppendixA-FAQ.md [hunk 35]
```diff
@@ -1 +1,69 @@
...
+- See also [re-running the Imaging pipeline](#rerunning-the-imaging-pipeline)
+    section for troubleshooting information.
+
+
+Feel free to visit the [FAQ](AppendixA-FAQ.md) section for more
```
Comment (cmadjar) [inline docs/AppendixA-FAQ.md]: isn't it the FAQ section? I would remove that last sentence.

File: docs/02-Install.md [hunk 36]
```diff
@@ -1,13 +1,311 @@
...
+ * `Path to Tarchives`: Directory where the original DICOMs are archived; typically `/data/$PROJECT/data/tarchive/`
+ * `Upload creation of candidates`: Enable or disable candidate creation into LORIS when running the insertion pipeline
+ * `Project batch management used`: Enable or disable batch management
+ * `If site is used`: Obsolete **To be confirmed**. This option used to allow the scans' incoming and archival paths to be configured based on the scanning sites
+ * `Number of volumes in native DTI acquisitions`: Used by the DTIPrep pipeline
+ * `Scan type of native T1 acquisition`: Name as specified in the `mri_scan_type` table. Used by the DTIPrep pipeline
+ * `Max number of DTI rejected directions for passing QC`: Number of directions that can be rejected and still pass QC. Used by the DTIPrep pipeline
+ * `NIAK Path`: Path to NIAK if MINC diffusion is to be run. Used by the DTIPrep pipeline
+ * `Secondary QCed dataset`: Path where a secondary QC'ed dataset it to be stored. Used by the DTIPrep pipeline
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md [hunk 37]
```diff
@@ -1,13 +1,311 @@
...
+
+
+### 2.2.2 LORIS
+
+1. **Imaging Uploader**
+
+
+Projects can upload scans and launch the pipeline in a variety of options 
+detailed in [PipelineOptions](PipelineOptions.md).
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: yes, it is because this needs to be rebased on the latest branch, but I am based off of yours...
I will attempt rebasing...

File: docs/02-Install.md [hunk 38]
```diff
@@ -1,13 +1,311 @@
...
+the [LORIS repository: Imaging Uploader Specification](https://github.com/aces/Loris/blob/master/modules/imaging_uploader/README.md).
+
+
+2. **DICOM Archive**
+
+This LORIS module provides a front-end display of the details of the archived 
+DICOMs from the database `tarchive_*` tables. The only setting that impacts the 
+display of this module are the regex settings in the `Configuration` module 
+under the section `Imaging Modules`. These settings determine whether the 
+Patient Name/Patient ID headers are displayed in full, or show up as
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md [hunk 39]
```diff
@@ -1,13 +1,311 @@
...
+
+This LORIS module provides a front-end display of the details of the archived 
+DICOMs from the database `tarchive_*` tables. The only setting that impacts the 
+display of this module are the regex settings in the `Configuration` module 
+under the section `Imaging Modules`. These settings determine whether the 
+Patient Name/Patient ID headers are displayed in full, or show up as 
+**INVALID-HIDDEN**.
+
+More detailed specifications can be consulted in the 
+[LORIS repository: DICOM Archive Specification](https://github.com/aces/Loris/blob/master/modules/dicom_arhive/README.md).
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: yes, I know. I made an earlier comment saying that these are currently on the minor branch, once they are released and forwarded to master, they should work...
not sure howelse to make this link.
I can set it to minor, if you want to test it, but I believe that master/ is the one we should be linking to. what do you think?

File: docs/02-Install.md [hunk 40]
```diff
@@ -1,13 +1,311 @@
...
+- Verify the Configuration module (*Paths*) `Imaging data`, `MINC files` and 
+  `Images` settings are set (typically: `/data/$PROJECT/data/`). 
+    
+More detailed specifications can be consulted in the 
+[LORIS repository: Imaging Browser Specification](https://github.com/aces/Loris/blob/master/modules/imaging_browser/README.md).
+
+4. **Brainbrowser**
+
+Brainbrowser accesses the MINC images directly from the filesystem where they
+are stored. Ensure that:
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok!

File: docs/AppendixA-FAQ.md (hunk 31 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: ok

File: docs/AppendixA-FAQ.md (hunk 32 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: ok

File: docs/AppendixA-FAQ.md (hunk 33 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: I will remove the link altogether

File: docs/AppendixA-FAQ.md (hunk 34 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: yes, but this is all from the wiki, and I think it depends on it was organized.
I removed the link altogether from the previous script.

File: docs/AppendixA-FAQ.md (hunk 35 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: lol, yes it is! removed

File: docs/02-Install.md [hunk 41]
```diff
@@ -1,13 +1,311 @@
...
+- `/data/$PROJ` directory and subdirectories are readable and executable by
+    the Apache linux user.
+- Verify the Configuration module (*Paths*) `MINC files` setting is
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok

File: docs/02-Install.md [hunk 42]
```diff
@@ -1,13 +1,311 @@
...
+    `/data/$PROJ/data/`.
+- Ensure the _project/config.xml_ file (in the main LORIS codebase) contains the
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok removed altogether and just said to make sure to have the proper MIncToolsPath set in the config.xml

File: docs/02-Install.md [hunk 43]
```diff
@@ -1,13 +1,311 @@
...
+[LORIS repository: DICOM Archive Specification](https://github.com/aces/Loris/blob/master/modules/dicom_arhive/README.md).
+
+
+3. **Imaging Browser**
+
+Imaging Browser accesses the PIC images directly from the filesystem where they
+are stored. It also provides the option to doownload some files.  Ensure that:
+- `/data/$PROJ` directory and subdirectories are readable and executable by
+    the Apache linux user.
+- Verify the Configuration module (*Paths*) `Imaging data`, `MINC files` and
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: done

File: docs/02-Install.md [hunk 44]
```diff
@@ -1,13 +1,311 @@
...
+    <!-- MINC TOOLS PATH -->
+    <MINCToolsPath>/opt/minc/</MINCToolsPath>
+    ```
+More detailed specifications can be consulted in the 
+[LORIS repository:Brainbrowser Specification](https://github.com/aces/Loris/blob/master/modules/brainbrowser/README.md).
+
+
+5. **MRI Violated Scans**
+
+No configuration setting is needed for the MRI Violated Scans LORIS module to
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok

File: docs/02-Install.md [hunk 45]
```diff
@@ -1,13 +1,311 @@
...
+[LORIS repository:Brainbrowser Specification](https://github.com/aces/Loris/blob/master/modules/brainbrowser/README.md).
+
+
+5. **MRI Violated Scans**
+
+No configuration setting is needed for the MRI Violated Scans LORIS module to 
+work. Data loaded in this module gets populated automatically by the insertion
+scripts. As such, scans whose parameters can't be matched against the 
+`mri_protocol` table during the imaging insertion process, will be flagged as 
+protocol violations and will not have their MINC/NIFTII volumes loaded in the
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: ok

File: docs/02-Install.md [hunk 46]
```diff
@@ -1,13 +1,311 @@
...
+/data/$projectname/data/
+/data/$projectname/bin/mri/
+/data/incoming/
+/data/$projectname/bin/mri/dicom-archive/.loris_mri/prod
+```
+
+#### 2.4.4 Verify Configuration module settings for Imaging Pipeline
+
+In the LORIS front-end, under the Admin menu, go to the `Configuration` module. 
+Under the section `Imaging Pipeline`, verify/set the following config settings:
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/02-Install.md]: lol; this comes from your pull request though :) but yeah, good idea to link them to the earlier section.

File: docs/04-Scripts.md [hunk 47]
```diff
@@ -1 +1,145 @@
...
+LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative 
+Neuroscience 
+
+=cut
+
+</pre>
+
+## 4.2 - Common insertion scripts re-run
+
+##### Bypassing protocol violation checks
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/04-Scripts.md]: lol can not believe you noticed this; unbelievable!

File: docs/04-Scripts.md [hunk 48]
```diff
@@ -1 +1,145 @@
...
+```
+uploadNeuroDB/minc_insertion.pl -acquisition_protocol t2w -bypass_extra_file_checks -create_minc_pics -profile prod -globLocation -force  -tarchivePath _/data/project/dataTransfer/library/2009/DCM_2009-09-25_project_20110214_185904581.tar_ -mincPath _/data/project/data/trashbin/TarLoad-3-34-pVzGC5/xxx0067_703739_v12_20090925_222403_18e1_mri.mnc_
+```
+
+Note carefully the following arguments:
+
+- *acquisition_protocol*: must be a known scan type according to the
+    `mri_scan_type` table
+- *tarchive_Path*: the DICOM tarball
+- *mincPath*: note this file may haven placed in the `trashbin` directory
```
[the remark this answers is not in the document]
Comment (MounaSafiHarab) [reply docs/04-Scripts.md]: I assume this is what @christinerogers meant to say in the wiki

File: docs/04-Scripts.md (hunk 23 above)
Comment (MounaSafiHarab) [reply docs/04-Scripts.md]: good question, me neither (just moved around your text that you got from the wiki). Removed the extra `_`

File: docs/04-Scripts.md (hunk 25 above)
Comment (MounaSafiHarab) [reply docs/04-Scripts.md]: I agree. This is how the wiki is currently...

File: docs/04-Scripts.md (hunk 26 above)
Comment (MounaSafiHarab) [reply docs/04-Scripts.md]: I did an attempt; hope it is clearer now.

File: docs/AppendixA-FAQ.md (hunk 27 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: ok

File: docs/AppendixA-FAQ.md (hunk 29 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: ok

File: docs/AppendixA-FAQ.md (hunk 30 above)
Comment (MounaSafiHarab) [reply docs/AppendixA-FAQ.md]: ok

**Comment by MounaSafiHarab** [comment]
@cmadjar 

rebasing was too painful, so I am closing this and sending a new one, but your comments raised here are already addressed (here and in the new one!).
